File: app/src/main/java/com/amplifyframework/datastore/sample/Modification.java
package com.amplifyframework.datastore.sample;

import com.amplifyframework.datastore.DataStoreItemChange;

enum Modification {
    CREATE, UPDATE, DELETE;

    static Modification from(DataStoreItemChange.Type type) {
        final Modification modification;
        switch (type) {
            case DELETE:
                return Modification.DELETE;
            case CREATE:
                return Modification.CREATE;
            case UPDATE:
                return Modification.UPDATE;
            default:
                throw new IllegalStateException("No modification known for " + type);
        }
    }
}

File: app/src/main/java/com/amplifyframework/datastore/sample/Modification.kt
package com.amplifyframework.datastore.sample

import com.amplifyframework.datastore.DataStoreItemChange

internal enum class Modification {
    CREATE, UPDATE, DELETE;

    companion object {
        fun from(type: DataStoreItemChange.Type): Modification {
            return when (type) {
                DataStoreItemChange.Type.DELETE -> DELETE
                DataStoreItemChange.Type.CREATE -> CREATE
                DataStoreItemChange.Type.UPDATE -> UPDATE
                else -> throw IllegalStateException("No modification known for $type")
            }
        }
    }
}
